iroh-gateway/src/client.rs

```diff
@@ -1,18 +1,27 @@
 use std::sync::Arc;
 
-use axum::body::Body;
+use axum::body::StreamBody;
 use cid::Cid;
+use iroh_resolver::resolver::OutPrettyReader;
+use iroh_resolver::resolver::Resolver;
+use tokio_util::io::ReaderStream;
 use tracing::info;
 
 use crate::core::State;
 use crate::response::ResponseFormat;
 
-#[derive(Debug, Clone, Copy)]
-pub struct Client {}
+#[derive(Debug)]
+pub struct Client {
+    resolver: Resolver<iroh_rpc_client::Client>,
+}
+
+pub type PrettyStreamBody = StreamBody<ReaderStream<OutPrettyReader<iroh_rpc_client::Client>>>;
```

Comment:
We might want to wrap this and hook in some metrics.

Reply:
as far as I understand axum, we probably want to add a generic middleware that does that for us